cognite/client/_api/postgres_gateway/users.py

```diff
@@ -0,0 +1,274 @@
+                >>> from cognite.client.data_classes import ClientCredentials
+                >>> client = CogniteClient()
+                >>> session = client.iam.sessions.create(
+                ...     ClientCredentials(os.environ.get("IDP_CLIENT_ID"), os.environ.get("IDP_CLIENT_SECRET")),
+                ...     session_type="CLIENT_CREDENTIALS"
+                ... )
+                >>> user = UserWrite(credentials=SessionCredentials(nonce=session.nonce))
+                >>> res = client.postgres_gateway.users.create(user)
+
+        """
+        self._warning.warn()
+        return self._create_multiple(
+            list_cls=UserList,
+            resource_cls=User,
+            items=user,
+            input_resource_cls=UserWrite,
+            headers={"cdf-version": "beta"},
+        )
+
+    @overload
+    def update(self, items: UserUpdate | UserWrite) -> User: ...
+
+    @overload
+    def update(self, items: Sequence[UserUpdate | UserWrite]) -> UserList: ...
+
+    def update(self, items: UserUpdate | UserWrite | Sequence[UserUpdate | UserWrite]) -> User | UserList:
+        """`Update users <https://api-docs.cognite.com/20230101-beta/tag/Postgres-Gateway-Users/operation/update_users>`_
+
+        Update postgres users
+
+        Args:
+            items (UserUpdate | UserWrite | Sequence[UserUpdate | UserWrite]): No description.
+
+        Returns:
+            User | UserList: The updated user(s)
+
+        Examples:
+
+            Update user:
+
+                >>> import os
+                >>> from cognite.client import CogniteClient
+                >>> from cognite.client.data_classes.postgres_gateway import UserUpdate, SessionCredentials
+                >>> from cognite.client.data_classes import ClientCredentials
+                >>> client = CogniteClient()
+                >>> session = client.iam.sessions.create(
+                ...     ClientCredentials(os.environ.get("IDP_CLIENT_ID"), os.environ.get("IDP_CLIENT_SECRET")),
+                ...     session_type="CLIENT_CREDENTIALS"
+                ... )
+                >>> client = CogniteClient()
+                >>> update = UserUpdate('myUser').credentials.set(SessionCredentials(nonce=session.nonce))
+                >>> res = client.postgres_gateway.users.update.update(update)
+
+        """
+        self._warning.warn()
+        return self._update_multiple(
+            items=items,
+            list_cls=UserList,
+            resource_cls=User,
+            update_cls=UserUpdate,
+            headers={"cdf-version": "beta"},
+        )
+
+    def delete(self, username: str | SequenceNotStr[str], ignore_unknown_ids: bool = False) -> None:
+        """`Delete postgres user(s) <https://api-docs.cognite.com/20230101-beta/tag/Postgres-Gateway-Users/operation/delete_users>`_
+
+        Delete postgres users
+
+        Args:
+            username (str | SequenceNotStr[str]): Username to authenticate the user on the DB.
```

Comment:
What about something more like "usernames of the users to delete"?

Reply:
Wdyt overwriting the first argument docs in retrieve and delete?

Reply:
> Agreed. Note this is coming from the spec.

Oh, I see... hm.. feel free to keep it, but I'd prefer something that helps explain a bit more with the given context (i.e. delete)